packages/cli/src/utils/dependencies.js

```diff
@@ -0,0 +1,49 @@
+import consola from 'consola'
+import { satisfies } from 'semver'
+
+import { getPKG } from '@nuxt/utils'
+
+export function checkDependencies() {
+  const packagesToTrack = ['postcss', 'webpack']
+  const webpackDeps = getPKG('@nuxt/webpack').dependencies
```

Comment:
An inline list would be more than enough since exceptions are manual anyway (and the fact that some deps are optional like sass-loader)

Reply:
I would use an object syntax like `{ 'sass-loader':  { range: '^10.1.1' }, 'node': { range: '>12', message: 'you are using unsupported NodeJS version' }`

Reply:
Node.js version checking has to be done differently so I think we don't need a single object.

Reply:
Actually it was idea of #8783 for generic version checking system. And we might make smarter warnings by checking if package is explicitly installed by `{rootDir}/package.json` or hoisted. But seems okay for now we might do it later on as a nuxt-contrib project :)